be/src/storage/lake/compaction_scheduler.cpp

```diff
@@ -157,8 +157,14 @@ CompactionScheduler::CompactionScheduler(TabletManager* tablet_mgr)
 }
 
 CompactionScheduler::~CompactionScheduler() {
-    _stopped.store(true, std::memory_order_relaxed);
-    _threads->shutdown();
+    stop();
+}
+
+void CompactionScheduler::stop() {
+    if (!_stopped) {
+        _stopped.store(true);
+        _threads->shutdown();
+    }
```

Comment:
```suggestion
    bool expected = false;
    if (_stopped.compare_exchange_strong(&expected, true)) {
        _threads->shutdown();
    }
```

make sure the check and store ops are atomic under no lock even though it is expected no concurrent call of this `stop()` interface.

Reply:
done